src/legacy/server/i18n/index.js

```diff
@@ -20,49 +20,66 @@
 import { resolve } from 'path';
 import globby from 'globby';
 import { i18n, i18nLoader } from '@kbn/i18n';
-
+import { readFile } from 'fs';
 import { fromRoot } from '../../utils';
+import { promisify } from 'util';
+
+const readFileAsync = promisify(readFile);
+
+export async function getTranslationPaths({ cwd, glob }) {
+  const entries = await globby(glob, { cwd });
+
+  const configFiles = await Promise.all(
+    entries
+      .map(entry => resolve(cwd, entry))
```

Comment:
I kept it separate intentionally, performance shouldn't be an issue here since the entries are few.

Reply:
Agree with @azasypkin, we should merge these, as there is another below. I think this could all be put into:

```js
.reduce(async (translationPaths, entry) => {
  const path = resolve(cwd, entry);

  try {
    return [
      ...translationPaths,
      ...JSON.parse(await readFileAsync(path, 'utf8'))
        .translations.map(translationPath => resolve(cwd, translationPath)),
    ];
  } catch (err) {
    throw new Error(`Failed to parse .i18nrc.json file at ${path}`);
  }
}, [])
```